src/state/opamSysInteract.ml

```diff
@@ -677,6 +698,11 @@ let install_packages_commands_t ?(env=OpamVariable.Map.empty) sys_packages =
     in
     [`AsAdmin "port", yes ["-N"] ("install"::packages)],
     None
+  | Msys2 pacman ->
+    (* NOTE: MSYS2 interactive mode may break (not show output until key pressed)
+       when called from opam. Confer
+       https://www.msys2.org/wiki/Terminals/#mixing-msys2-and-windows. *)
+    [`AsUser pacman, "-Su"::"--noconfirm"::packages], None
```

Comment:
Discussed offline that this isn't ideal, but it is necessary because of MSYS2 terminal and somewhat safe because MSYS2 changes don't break Windows (unlike other package managers).